### PR TITLE
[ADS] Create NewHandler function, tests, and README (DMX-615)

### PR DESCRIPTION
**Pull request by reallyangrydude**
Added NewHandler basic functionality.  Created readme file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    Secret: "5d63GMY5fRsBRdB7cDsMoLlNX9vWxNSq",
+    Issuer: "dmissuer",
+    Audiences: []string{"dm"},
+    BasicUser: "dmxadmin",
+    BasicPass: "xhunterx",
+    Skip: []string{"/healthz"},
+}
+
+func tokenGetter(r *http.Request) (string, error) {
+    return "", nil
```
Comment (kalbasit) [inline README.md]: I believe it's better to have a real example here.

```
v := r.URL.Values()
if t, ok := v["access_token"]; ok {
    return t, nil
} else {
    return "", errors.New("not found")
}
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+c := xhandler.Chain{}
+c.UseC(xjwt.NewHandler(c, tokenGetter))
+```
+
+See [XHandler](https://github.com/rs/xhandler) for how to use Xhandler.
+
+Licenses
+--------
+All source code is licensed under the 
+[MIT License](https://raw.githubusercontent.com/rs/xhandler/master/LICENSE).
```
Comment (kalbasit) [inline README.md]: Do not refer to xhandler's License here. Copy the `LICENSE` over from [forensiq](https://github.com/dailymotion/forensiq/blob/master/LICENSE) to this repository and link to it.

File: xjwt.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+func NewHandler(c Config, f func(*http.Request) (string, error)) func(xhandler.HandlerC) xhandler.HandlerC {
+	return func(next xhandler.HandlerC) xhandler.HandlerC {
+		return xhandler.HandlerFuncC(func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+			var auth bool
+
+			// Skip verification of path is in skip list.
+			for _, skipPath := range c.Skip {
+				if r.URL.Path == skipPath {
+					auth = true
```
Comment (kalbasit) [inline xjwt.go]: add a `break` here. No need to continue looping over all the skipPaths when you find that it needs to be skiped.

File: xjwt.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+				tok, err := f(r)
+				if err == nil {
+					// Check token credentials.
+					auth = checkToken(c, tok)
+				} else {
+					http.Error(
+						w,
+						fmt.Sprintf("error getting the token from the request: %v", err),
+						http.StatusInternalServerError,
+					)
```
Comment (kalbasit) [inline xjwt.go]: you have to return here. Otherwise, you will send an HTTP error but keep the logic going...

File: README.md (hunk 1 above)
Comment (rs) [inline README.md]: Why do you need to provide a token getter here? The default behavior should be to get the bearer token in the header and to fallback on the `access_token` POST or GET parameter as defined in the oauth2 spec: https://tools.ietf.org/html/rfc6750#section-2

Comment (kalbasit) [inline README.md]: Oops, my mistake, I didn't read the entire [`jwt.ParseFromRequest`](https://github.com/dgrijalva/jwt-go/blob/master/token.go#L94-L112). @reallyangrydude just use ParseFromRequest then,

**Comment by kalbasit** [comment]
LGTM

**Pull request closed without merging**